sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_patch.py

```diff
@@ -95,16 +196,28 @@ async def publish_cloud_events(
         :rtype: None
         :raises ~azure.core.exceptions.HttpResponseError:
         """
+        
+        # Check that the body is a CloudEvent or list of CloudEvents even if dict
+        if isinstance(body, dict) or (isinstance(body, list) and isinstance(body[0], dict)):
+            try:
+                if isinstance(body, list):
+                    body = [CloudEvent.from_dict(event) for event in body]
+                else:
+                    body = CloudEvent.from_dict(body)
+            except AttributeError:
+                raise TypeError("Incorrect type for body. Expected CloudEvent,"
+                                " list of CloudEvents, dict, or list of dicts."
+                                " If dict passed, must follow the CloudEvent format.")
+
+
         if isinstance(body, CloudEvent):
             kwargs["content_type"] = "application/cloudevents+json; charset=utf-8"
-            internal_body = _cloud_event_to_generated(body)
-            await self._publish_cloud_event(topic_name, internal_body, **kwargs)
-        else:
+            await self._publish(topic_name, body, self._config.api_version, binary_mode, **kwargs)
+        elif isinstance(body, list):
             kwargs["content_type"] = "application/cloudevents-batch+json; charset=utf-8"
-            internal_body_list = []
-            for item in body:
-                internal_body_list.append(_cloud_event_to_generated(item))
-            await self._publish_cloud_events(topic_name, internal_body_list, **kwargs)
+            await self._publish(topic_name, body, self._config.api_version, binary_mode, **kwargs)
+        else:
+            raise TypeError("Incorrect type for body. Expected CloudEvent or list of CloudEvents.")
```

Comment:
perhaps add in list of dicts here so that user gets a consistent error message around proper types